PR #5: Daily / weekly toggle for events

KamilaKinel opened the pull request:
# Description

The following PR includes the following:
-  `Toggle` component above accordion for the user to quickly switch between daily and weekly 
-  Events component that displays events
- Updated endpoint calls to supply either daily or weekly 
- Marked where additional filtering should be done 
- Tests for both


## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Has been manually tested and tested using written tests as well.  Test coverage can be seen below:
![Screenshot 2025-04-20 at 3 44 09 PM](https://github.com/user-attachments/assets/24c16dd2-fbfd-449d-9198-867126b717f1)
![Screenshot 2025-04-20 at 3 43 52 PM](https://github.com/user-attachments/assets/17257751-3d47-4ca8-baaf-688a5c84d1a3)


- [x] Manual Testing
- [x] Automated Unit Testing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# UML Diagram

<img width="873" alt="Screenshot 2025-04-21 at 11 09 09 AM" src="https://github.com/user-attachments/assets/fa89a8bd-fda3-475f-9d02-557523a0c746" />

daltog1 reviewed (changes requested):
Awesome work on this Kamila! I think once everything is ironed out on this it will give users a great option to have for their popup UI. We already discussed incorporating the Upcoming Assignments into this daily/weekly toggle so I am excited to see that come together. Here's a couple things I experienced when testing this PR:

- Weekdays and actual dates on the weekly view did not line up. One example of this is on my "Your Events" list I see this listed at 12:00 PM on Sunday. 
![image](https://github.com/user-attachments/assets/a001d14e-20a0-444c-a7fc-3389ace9a635)

But when I click on the link, it is actually at 12:00 PM tomorrow (4/21) 
![image](https://github.com/user-attachments/assets/8e0d7803-a642-47e7-a7c3-0e3a06c86c44)

This seems to be the case for the rest of the days of the week as well. I'm not sure if that is a database endpoint issue or the way the events are handled but please look into this so the correct events are displayed. On the daily view, the correct events for today are shown. 

- Could we please organize the events so they are listed by start time? When I looked at my "Your Events" list I saw events in a hard to follow order. 
![image](https://github.com/user-attachments/assets/b03ba7e8-33a1-4a32-b73a-22b094fdc749)

By adjusting this, users would be less likely to miss something because they did not have to scroll through the entire list, only to the time they were looking for.

- Would it be a good idea to not show any events that are past their end time? For example, today was an Easter egg hunt that started at 11:00 AM and has an end time of 1:00 PM. 
![image](https://github.com/user-attachments/assets/6f8d1668-b9e1-4314-bae2-cf865b4821d7)

Would I still want to see the old events from today that are already completed? This one was just a thought I had but I wanted to see how everyone would like to handle this. 

This PR looks great and will work great for the user. Awesome job adding tests for the files you added. Keep up the great work!

KamilaKinel commented:
@daltog1 Yes good catch! It was adding in the order it was in the database.  I added something to sort by times.  

Also very good catch with the dates, was a weird error.  It is also now fixed.

Please retest when you have the chance :)

StarStiles reviewed (approved):
Great work, Kamila! Love the way that you displayed events in the accordion menu, and the daily/weekly toggle works great! We'll use this as a framework for formatting our events, and once we add the filtering for all the events we'll be all set!

SebastianYepez reviewed (approved):
Wonderful PR!

I love the fact that you saved the toggle state to storage to keep track of whether or not the user last chose to see their events on "Daily" or "Weekly" view. Also, I love that the day of the week we are currently on is the first shown, so since today is Monday the 21st, it shows all of today's events up to Sunday the 27th's. Very detailed PR and this is exactly what I imaged our project looking like!


![image](https://github.com/user-attachments/assets/fb933029-9602-4f13-973d-e5371754e1a0)
![image](https://github.com/user-attachments/assets/b9c10659-04d1-4a11-b32d-d7a3a8b82a88)

daltog1 reviewed (approved):
This looks great now! Everything I requested has been added so I will approve.
